packages/zql/src/query/named.test.ts

```diff
@@ -27,64 +29,141 @@ test('defining a named query', () => {
       }[]
     >
   >();
-  check(x.myName);
 
-  // define many at once
-  const y = queries({
-    myName: (id: string) => queryBuilder.issue.where('id', id),
-    myOtherName: (id: string) => queryBuilder.issue.where('id', id),
-    myThirdName: (id: string) => queryBuilder.issue.where('id', id),
+  expect(q.customQueryID).toEqual({
+    name: 'myQuery',
+    args: ['123'],
+  });
+
+  const defWithFakeContext = withContext(def);
+  q = defWithFakeContext('1', '321');
+  expect(q.customQueryID).toEqual({
+    name: 'myQuery',
+    args: ['321'],
   });
-  check(y.myName, 'myName');
-  check(y.myOtherName, 'myOtherName');
-  check(y.myThirdName, 'myThirdName');
-  const q1 = y.myName('123');
-  const q2 = y.myOtherName('123');
-  const q3 = y.myThirdName('123');
-  expectTypeOf<ReturnType<typeof q1.run>>().toEqualTypeOf<
-    ReturnType<typeof q.run>
+
+  // no validator was defined
+  expect(() => withValidation(def)).toThrowErrorMatchingInlineSnapshot(
+    `[Error: ret does not have a validator defined]`,
+  );
+});
+
+test('defining a synced query with context', () => {
+  const def = syncedQueryWithContext('myQuery', (_c: unknown, id: string) =>
+    builder.issue.where('id', id),
+  );
+  const q = def(1, '123');
+  expectTypeOf<ReturnType<typeof q.run>>().toEqualTypeOf<
+    Promise<
+      {
+        readonly id: string;
+        readonly title: string;
+        readonly description: string;
+        readonly closed: boolean;
+        readonly ownerId: string | null;
+        readonly createdAt: number;
+      }[]
+    >
   >();
-  expectTypeOf<ReturnType<typeof q2.run>>().toEqualTypeOf<
-    ReturnType<typeof q.run>
+
+  expect(q.customQueryID).toEqual({
+    name: 'myQuery',
+    args: ['123'],
+  });
+
+  // no validator was defined
+  expect(() => withValidation(def)).toThrowErrorMatchingInlineSnapshot(
+    `[Error: ret does not have a validator defined]`,
+  );
+});
+
+test('defining a synced query with validation', () => {
+  const def = syncedQuery(
+    'myQuery',
+    (id: unknown) => {
+      assert(typeof id === 'string', 'id must be a string');
+      return [id] as const;
+    },
+    (id: string) => builder.issue.where('id', id),
```

Comment:
Can you add a test that the inference works? Should not need to provide type on `id` in query func.